crates/ruff_python_formatter/src/comments/format.rs

```diff
@@ -294,51 +318,10 @@ pub(crate) struct FormatComment<'a> {
 
 impl Format<PyFormatContext<'_>> for FormatComment<'_> {
     fn fmt(&self, f: &mut PyFormatter) -> FormatResult<()> {
-        let slice = self.comment.slice();
-        let comment_text = slice.text(SourceCode::new(f.context().source()));
-
-        let trimmed = comment_text.trim_end();
-        let trailing_whitespace_len = comment_text.text_len() - trimmed.text_len();
-
-        let Some(content) = trimmed.strip_prefix('#') else {
-            return Err(FormatError::syntax_error(
-                "Didn't find expected comment token `#`",
-            ));
-        };
-
-        // Fast path for correctly formatted comments:
-        // * Start with a `#` and are followed by a space
-        // * Have no trailing whitespace.
-        if trailing_whitespace_len == TextSize::new(0) && content.starts_with(' ') {
-            return source_text_slice(slice.range(), ContainsNewlines::No).fmt(f);
-        }
-
-        write!(f, [source_position(slice.start()), text("#")])?;
-
-        // Starts with a non breaking space
-        let start_offset =
-            if content.starts_with('\u{A0}') && !content.trim_start().starts_with("type:") {
-                // Replace non-breaking space with a space (if not followed by a normal space)
-                "#\u{A0}".text_len()
-            } else {
-                '#'.text_len()
-            };
-
-        // Add a space between the `#` and the text if the source contains none.
-        if !content.is_empty() && !content.starts_with([' ', '!', ':', '#', '\'']) {
-            write!(f, [space()])?;
-        }
+        let source = SourceCode::new(f.context().source());
+        let comment = normalize_comment(self.comment, source)?;
 
-        let start = slice.start() + start_offset;
-        let end = slice.end() - trailing_whitespace_len;
-
-        write!(
-            f,
-            [
-                source_text_slice(TextRange::new(start, end), ContainsNewlines::No),
-                source_position(slice.end())
-            ]
-        )
+        dynamic_text(&comment, None).fmt(f)
```

Comment:
Using `dynamic_text` for both `Cow::Borrowed` and `Cow::Owned` nullifies the benefit of using a `Cow`, because it always allocates a string internally. 

What we want to do here is:

```
match comment {
	Cow::Borrowed(borrowed) => source_text_slice(TextRange::at(start, borrowed.text_len(), ContainsNewlines::No),
	Cow::Owned(owned) => dynamic_text(&owned, Some(start))
}
```

Reply:
Ty!

https://github.com/astral-sh/ruff/compare/0aa3a86f1f641c9e67b0efa9f57924a038bb2eef..2f0f844005fa108fe627b21fab62cd1b04fd2c8e